### PR TITLE
Add 1.7.1 to 1.8.0 update guide (i.e. everything but Focal)

### LINKED ISSUE #145: Revise tails3-4 migration docs
User reports that securedrop-admin setup fails during the apt stage on Tails 3 now with GPG key errors on the Tor repos. This makes sense as their key was updated recently. It also invalidates the instructions that say to upgrade to the latest SD version first before updating from Tails 3 to 4. They should be revised.

### PR DESCRIPTION
**Pull request by eloquence**
## Status
Ready for final review. Per [RM guide](https://docs.securedrop.org/en/stable/development/release_management.html), marked as draft until release is live (step 20 of the docs) and we're ready to merge.

## Description of Changes

Part of https://github.com/freedomofpress/securedrop/issues/5794.
Resolves #145.

- Begins the process of removing Tails 3->Tails 4 docs. Once no release upgrade guide references them, we can remove them completely.
- Per standard operating procedure, removes the oldest release upgrade guide.
- Adds Tails 4.14  upgrade breakage advisory, which we probably need to carry forward for a while.
- Bumps the docs version to 1.8.0.

## Testing
- No special instructions.

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

File: docs/upgrade/1.7.1_to_1.8.0.rst
Comment (eloquence) [inline docs/upgrade/1.7.1_to_1.8.0.rst:9]: Link TK

File: docs/upgrade/1.7.1_to_1.8.0.rst
Comment (eloquence) [inline docs/upgrade/1.7.1_to_1.8.0.rst:89]: Link TK

File: docs/upgrade/1.7.1_to_1.8.0.rst
Comment (eloquence) [inline docs/upgrade/1.7.1_to_1.8.0.rst:11]: Heading slightly altered (usually it is "Automatic server upgrades") to mitigate risk of confusion between Ubuntu/SecureDrop version updates.

**Review by rocodes — COMMENTED** [review]
LVGTM. As soon as #169 is merged this will be good to go.

File: docs/upgrade/1.7.1_to_1.8.0.rst
Comment (rocodes) [reply docs/upgrade/1.7.1_to_1.8.0.rst:11]: j'approuve

**Review by rocodes — APPROVED** [review]
:100:  
Looks good to merge once release is live

**Pull request merged**
